src/components/Requirements.vue

```diff
@@ -508,19 +512,46 @@ export default {
        * @param {*} search : the scope of search for the requirement (e.g all-eligible, code, catalogDistr)
        * @param {*} includes : the query for the search (e.g (MQR-AS), CS 2***)
        */
-      function checkIfCourseFulfilled(courseInfo, search, includes) {
-        // Special search: if search code is all or self-check. Anything would work
-        if (search === 'all' || search === 'self-check') return true;
-        // Special search: if search code is not PE or 10XX course
-        if (search === 'all-eligible') return ifAllEligible(courseInfo.subject, courseInfo.catalogNbr.toString());
-        for (const include of includes) {
-          for (const option of include) {
-            // Special search: if course code matches code
-            if (search === 'code') {
-              if (ifCodeMatch(`${courseInfo.subject} ${courseInfo.catalogNbr}`, option)) {
-                return true;
+      function checkIfCourseFulfilled(courseInfo, search, includes, excludes) {
+        // Check if search exists. False if not
+        if (search !== undefined) {
```

Comment:
Should this be similar to the comment you made to Theresa? if(search) to ensure its not undefined OR null.

Reply:
In this case it's specifically undefined. It will never be null or false, especially with the JSON schema validator. This way it's just more specific